doc/Language-Reference.adoc

```diff
@@ -0,0 +1,1625 @@
+
+*Syntax*
+
+[subs="+macros,quotes"]
+----
+[[syntax-variable_declaration]]
+variable_declaration ::= variable_xref:syntax-name[name] *:* type_xref:syntax-name[name] [ *:=* initialization_xref:syntax-expression[expression] ]
+----
+
+*Static Semantics*
+
+[%collapsible]
+====
+Types [§S-D-V-T]:
+
+* Scalar [§S-D-V-T-SC]
+* Message [§S-D-V-T-M]
+* Scalar Sequence [§S-D-V-T-SS]
+* Message Sequence [§S-D-V-T-MS]
+
+Initialization expressions [§S-D-V-E]:
+
+* No initialization [§S-D-V-E-N]
+* Mathematical Expressions [§S-D-V-E-ME]
+* Literals [§S-D-V-E-L]
+* Variables [§S-D-V-E-V]
+* Message Aggregates [§S-D-V-E-MA]
+* Aggregates [§S-D-V-E-A]
+* Valid Attributes [§S-D-V-E-VAT]
+* Opaque Attributes [§S-D-V-E-OAT]
+* Head Attributes [§S-D-V-E-HAT]
+* Has_Data Attributes [§S-D-V-E-HDAT]
+* Selected Expressions [§S-D-V-E-S]
+* List Comprehensions [§S-D-V-E-LC]
+* Bindings [§S-D-V-E-B]
+* Quantified Expressions [§S-D-V-E-Q]
+* Calls [§S-D-V-E-CL]
+* Conversions [§S-D-V-E-CV]
+====
+
+*Example*
+
+[source,ada,rflx,declaration]
+----
+Error_Sent : Boolean := False
+----
+
+===== Renaming Declaration [§S-D-R]
+
+*Syntax*
+
+[subs="+macros,quotes"]
+----
+[[syntax-renaming_declaration]]
+renaming_declaration ::= xref:syntax-name[name] *:* message_type_xref:syntax-name[name] *renames* message_variable_xref:syntax-name[name] *.* field_xref:syntax-name[name]
+----
+
+*Example*
+
+[source,ada,rflx,declaration]
+----
+Client_Hello_Message : TLS_Handshake::Client_Hello renames Client_Hello_Handshake_Message.Payload
+----
+
+==== States [§S-S]
+
+A state defines the to be executed actions and the transitions to subsequent states.
+
+*Syntax*
+
+[subs="+macros,quotes"]
+----
+[[syntax-state]]
+state ::=
+   *state* xref:syntax-name[name]
+    [ *with* xref:syntax-description_aspect[description_aspect] ]
+   *is*
+    { xref:syntax-state_declaration[state_declaration] }
+   *begin*
+    { xref:syntax-state_action[state_action] }
+   *transition*
+    { xref:syntax-conditional_transition[conditional_transition] }
+      xref:syntax-transition[transition]
+ [ *exception*
+     xref:syntax-transition[transition] ]
+   *end* xref:syntax-name[name]
+ | *state* xref:syntax-name[name] *is null state*
+[[syntax-description_aspect]]
+description_aspect ::= *Desc =>* xref:syntax-string[string]
+----
+
+*Static Semantics*
+
+An § exception transition [§S-S-E] must be defined just in case any action might lead to a critical (potentially non-recoverable) error:
+
+* Insufficient memory for setting a field of a message
+* Insufficient memory for appending an element to a sequence or extending a sequence by another sequence
+
+Exception transitions are currently also used for other cases. This behavior will change in the future (cf. https://github.com/Componolit/RecordFlux/issues/569[#569]).
+
+A § null state [§S-S-N] does not contain any actions or transitions, and represents the final state of a session state machine.
```

Comment:
It's a separator, so that only "null state" is associated with "[§S-S-N]" (and not "A null state"). Do you think that is a big issue? We could also consider moving all these tags into [AsciiDoc comments](https://docs.asciidoctor.org/asciidoc/latest/comments/) to improve the readability.

Reply:
I found it confusing tbh. How much effort would it be to move the meta data into AsciiDoc comments?

Reply:
As discussed, we will do that in a separate issue: #715.